doc/api/buffer.md

```diff
@@ -197,6 +197,38 @@ the WHATWG specification it is possible that the server actually returned
 `'win-1252'`-encoded data, and using `'latin1'` encoding may incorrectly decode
 the characters.
 
+### Evaluating legal code points for '`utf-8'` encoding
+
+Byte sequences that do not have corresponding UTF-16 encodings and non-legal
+Unicode values, along with their UTF-8 counterparts must be treated as
+invalid byte sequences.
+
+For cases regarding operations other than employing backward compatibility
+for 7-bit (and [extended 8-bit]((https://en.wikipedia.org/wiki/UTF-8#Description))
+in rare cases) `'ascii'` data, and the valid [`UTF-8` code units](https://en.wikipedia.org/wiki/UTF-8#Codepage_layout),
+the replacement character (`�`) is returned,
+and no exception will be thrown.
+
+A `U+FFFD` replacement value
+(representing the aforementioned replacement character) will be returned
+in case of decoding errors (invalid unicode scalar values).
```

Comment:
To be honest, I don’t understand most of the text or its relevance here… the text basically says that invalid UTF-8 byte sequences will be decoded into `U+FFFD` replacement characters and that no error will be thrown in those cases, right?

How do UTF-16 and ASCII relate to that? What does “non-legal Unicode value” mean? (I would *guess* that this refers to characters that would be beyond U+10FFFF – if that’s correct, can you clarify that in the text?)